### ISSUE TITLE
Updating self-referenced many-to-many associated model causes invalid create calls

### ISSUE THREAD

**Issue by murhum1** [balderdashy/sails#3185]
Sails version 0.10.5, using sails-mysql adapter if it matters.

I have a model called `Organization`, which has the following fields:

    module.exports = {

	    attributes: {
			name: {
				type: 'String',
				unique: true,
				required: true,
				primaryKey: true
			},
		    contactPerson: {
				type: 'String'
			}
			members: {
				collection: 'User'
			},
			accessTo: {
				collection: 'Organization',
				via: 'accessedBy',
				dominant: true
			},
			accessedBy: {
				collection: 'Organization',
				via: 'accessTo'
			}
		}
    };

The `accessTo` and `accessedBy` indicate that members of an organization can access the data of organizations saved in its `accessTo` attribute, and the `accessedBy` is there just to be the other end of the association. I also have `beforeCreate` and `beforeUpdate` functions defined as follows:

	beforeCreate: function(values, cb) {
		console.log("At beforeCreate! Trying to create organization:", values.name);
		console.log("VALUES:", values);
		cb();
	},

	beforeUpdate: function(values, cb) {
		console.log("At beforeUpdate! Trying to update organization:", values.name);
		console.log("VALUES:", values);
		cb();
	}

Now, when trying to update the contactPerson attribute of an organization that can access two other organizations, it's for some reason firing `insert` commands to create both of the models the organization is associated with. The log statements output the following:

	At beforeCreate! Trying to create organization: AB Super City Illumination OY
	VALUES: { name: 'AB Super City Illumination OY',
	  contactPerson: null,
	  createdAt: Wed Aug 26 2015 11:08:05 GMT+0300 (EEST),
	  updatedAt: Wed Aug 26 2015 11:08:05 GMT+0300 (EEST) }

	At beforeCreate! Trying to create organization: Mega Corp
	VALUES: { name: 'Mega Corp',
	  contactPerson: null,
	  createdAt: Wed Aug 26 2015 11:08:05 GMT+0300 (EEST),
	  updatedAt: Wed Aug 26 2015 11:08:05 GMT+0300 (EEST) }

Then, because the organizations already exist, I get the error:

	{ type: 'insert',
	    collection: 'organization',
	    values: 
	     { name: 'AB Super City Illumination OY',
	       contactPerson: null,
	       createdAt: '2015-08-26T08:08:05.000Z',
	       updatedAt: '2015-08-26T08:08:05.000Z' },
	    err: 
	     Error (E_VALIDATION) :: 1 attribute is invalid
	         at WLValidationError.WLError (/home/markus/Documents/manager/node_modules/sails/node_modules/waterline/lib/waterline/error/WLError.js:26:15)
	         at new WLValidationError (/home/markus/Documents/manager/node_modules/sails/node_modules/waterline/lib/waterline/error/WLValidationError.js:20:28)
	         at duckType (/home/markus/Documents/manager/node_modules/sails/node_modules/waterline/lib/waterline/error/index.js:60:12)
	         at errorify (/home/markus/Documents/manager/node_modules/sails/node_modules/waterline/lib/waterline/error/index.js:39:10)
	         at wrappedCallback (/home/markus/Documents/manager/node_modules/sails/node_modules/waterline/lib/waterline/utils/normalize.js:325:15)
	         at _normalizeCallback.callback.error (/home/markus/Documents/manager/node_modules/sails/node_modules/waterline/node_modules/switchback/lib/normalize.js:42:31)
	         at _switch (/home/markus/Documents/manager/node_modules/sails/node_modules/waterline/node_modules/switchback/lib/factory.js:46:28)
	         at sendBackError (/home/markus/Documents/manager/node_modules/sails-mysql/lib/connections/spawn.js:91:11)
	         at Object.module.exports.poolfully [as releaseConnection] (/home/markus/Documents/manager/node_modules/sails-mysql/lib/connections/release.js:28:12)
	         at /home/markus/Documents/manager/node_modules/sails-mysql/lib/connections/spawn.js:90:37
	         at Query._callback (/home/markus/Documents/manager/node_modules/sails-mysql/lib/adapter.js:400:27)
	         at Query.Sequence.end (/home/markus/Documents/manager/node_modules/sails-mysql/node_modules/mysql/lib/protocol/sequences/Sequence.js:96:24)
	         at Query.ErrorPacket (/home/markus/Documents/manager/node_modules/sails-mysql/node_modules/mysql/lib/protocol/sequences/Query.js:94:8)
	         at Protocol._parsePacket (/home/markus/Documents/manager/node_modules/sails-mysql/node_modules/mysql/lib/protocol/Protocol.js:271:23)
	         at Parser.write (/home/markus/Documents/manager/node_modules/sails-mysql/node_modules/mysql/lib/protocol/Parser.js:77:12)
	         at Protocol.write (/home/markus/Documents/manager/node_modules/sails-mysql/node_modules/mysql/lib/protocol/Protocol.js:39:16)
	     
	     Invalid attributes sent to undefined:
	      • PRIMARY
	        • A record with that `PRIMARY` already exists (`AB Super City Illumination OY`).
	      }

What's going on in here? Surely on an update statement like this, no new models should be created? And still somewhere out of the blue these create commands are issued.

**Comment by murhum1** [comment]
Update: Seems like the `update` call is somehow deleting the association. The `accessTo` attribute becomes an empty array after trying to update. :|

**Comment by sailsbot** [comment]
Thanks for posting, @murhum1.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
